### ISSUE TITLE
General Discussion

### ISSUE THREAD

**Issue by lindseydiloreto** [lindseydiloreto/craft-businesslogic#1]
Hey guys, this is just a general discussion thread to figure out what works, what doesn't work, and suggestions for changes. This is a completely open forum, so feel free to bring up anything!

And feel free to fork this project! Pull requests are always welcome. :)

**Comment by lindseydiloreto** [comment]
First concern... I feel like the Variable and Service files are pretty solid (I even tested them). But the **Controller file** probably needs a bit of work. It's hard describing all of the things that a Controller can do, plus provide various examples. So if you have any suggestions, I'm all ears!

**Comment by joshangell** [comment]
@lindseydiloreto just wanted to say I think this is a brilliant idea! We frequently build a tiny plugin to hold the complex bits for a site and I never thought of making a template to start from. Kudos!

**Comment by lindseydiloreto** [comment]
@joshangell Thanks! I was sorta surprised at how many people quickly starred it on Github. :)

If you have any ideas on ways to make it better, forks and suggestions are welcome!

**Comment by mattstein** [comment]
@lindseydiloreto Ditto @joshangell's comments. I've wanted this for every plugin I've started, and now it exists and is amply described from within!

**Comment by ssssct** [comment]
@lindseydiloreto This is IMHO the best resource to get started with plugin dev for Craft. Now it would be great if it was ported to craft 3...

**Comment by lindseydiloreto** [comment]
Thanks @ssssct! I do have plans to port it to Craft 3... I may even turn it into a _module_ instead.

**Comment by codecreate** [comment]
Can it currently be used with Craft 3 at all?

**Comment by lindseydiloreto** [comment]
Not yet, but I've finally gotten around to porting it. Should be ready within the next few weeks. 👍

**Comment by lindseydiloreto** [comment]
I finally finished porting the Business Logic plugin template over to Craft 3, and actually converted it to a Business Logic **module** template instead. In Craft 3, it makes far more sense that someone would need this architecture as a module instead of a plugin.

**Business Logic is dead. Long live Business Logic!**  👑 

This time, I did an even deeper dive into the included components (Variables, Controllers, Services), writing up the purpose of each piece, and how they all tie together in the big picture.

https://doublesecretagency.github.io/craft-businesslogic/

And now, almost five years later, I feel comfortable closing this thread. 🙂
